runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordState.java

```diff
@@ -209,4 +209,11 @@ void checkForTimeout()
 		final Instant now = Instant.now();
 		events.removeIf(event -> event.getType().isShouldTimeout() && now.isAfter(event.getUpdated().plus(actionTimeout)));
 	}
-}
+
+	public void checkForMenuTimeout() {
+		EventWithTime event = events.get(0);
+		if (event.getType().getState() == DiscordGameEventType.IN_MENU.getState())
+			if (event.start.plusSeconds(900).isBefore(Instant.now()))
```

Comment:
The `900` should be a static constant with a descriptive name. See other plugins for examples of how to format it.